samples/Event_Jsvm/app/application.cpp

```diff
@@ -1,41 +1,109 @@
 #include <SmingCore.h>
-#include <Jsvm.h>
-#include <vm_functions.h>
+#include <Jerryscript.h>
 
 namespace
 {
-
-Jsvm jsVm;
-Timer tempTimer;
+JS::VirtualMachine vm;
+SimpleTimer timer;
+HashMap<String, JS::Callable::List> events;
 
 IMPORT_FSTR(main_snap, PROJECT_DIR "/out/jerryscript/main.js.snap")
 
+namespace JS
+{
+using namespace Jerryscript;
+
+/**
+ * @brief Function to register event listeners
+ * @code {.javascript}
+ * 
+ * event = {
+ *    name => "TEMP_CHANGE",
+ *    params = {
+ *    },
+ *    "origin" => "The\Creator\Of\The\Event" 
+ * };
+ * 
```

Comment:
The "origin" property sticks out a bit here. Weird value too! Purpose?

Reply:
There idea is the following: If you have two or more temperature sensors may be you would like to know which one sent the event. And the "origin" property can be used to provide that information. The value of the property is free text so it can be just `temp1` or `cpu/temp` and `extern/temp1`.